exercises/concept/go/.docs/instructions.md

```diff
@@ -0,0 +1,43 @@
+# Instructions
+
+In this exercise, you'll be applying the [rules of the game of Go](https://matmoore.github.io/learngo/). The rules themselves are already written, you just need to apply them in order to update the Game and to handle any errors / violations of the rules.
+
+The game is represented as follows:
+
+```elm
+type Player
+    = Black
+    | White
+
+
+type alias Game =
+    { whiteCapturedStones : Int
+    , blackCapturedStones : Int
+    , player : Player
+    , error : String
+    }
+```
+
+There are 4 rules in the game:
+
+- Each point can only have one stone (oneStonePerPointRule)
+- Opposition stones can be captured! (captureRule)
+- You can not place a stone where it would capture itself (libertyRule)
+- You can not use the same point twice (koRule)
+
+## 1. Apply the rules
```

Comment:
I think this exercise is a little bit too much in one go and could benefit from being split into 3 or 4 steps.

 Maybe the student could apply all these rules one at a time. Like `applyOneStonePerPointRule : Game -> (Game -> Result String Game) -> Result String Game ` and so on. Also, implementing `applyError : Game -> (Game -> Result String Game) -> Game` would be a nice extra step to practice pattern matching.

I also thing the `Rule` type alias might be too much abstraction. In an exercise concept about `Result`, it's OK to see `Result` everywhere. Maybe defining it just before the final `applyRules` would be OK.

I also think I would help to see a `board : Dict (Int, Int) (Maybe Player)` in `Game` or something just to make it look more like a game of Go?

Reply:
It is quite a chunky exercise, but you end up with something that looks like good real world code, and requires the use of all the common Result functions / syntax. If you have thought of a nice way of splitting it in to small steps while keeping all this please suggest a change / pr for the example code.

I don't think I understand applying the rules once at a time. That is basically just calling the function I think?

We could remove the rule abstraction, but the type annotations would become very big, and adding a type alias is something you would do in production code, so I think its ok. Maybe we could add a comment or something to the instructions to explain it in more detail (although `type alias` is a requirement of this exercise I think)

Adding a `board` isn't required for the exercise, so potentially makes things more complex, but maybe we could add it with a comment, or add it as a comment?

Reply:
I'll see if I can come up with something that works...